service/storage_service.cc

```diff
@@ -456,9 +456,14 @@ future<> storage_service::join_token_ring(cdc::generation_service& cdc_gen_servi
     // Seed the host ID-to-endpoint map with our own ID.
     auto local_host_id = _db.local().get_config().host_id;
     if (!replacing_a_node_with_diff_ip) {
-        // Replacing node with a different ip should own the host_id only after
-        // the replacing node becomes NORMAL status. It is updated in
-        // handle_state_normal().
+        auto endpoint = get_broadcast_address();
+        auto eps = _gossiper.get_endpoint_state_for_endpoint_ptr(endpoint);
+        if (eps) {
+            auto replace_host_id = _gossiper.get_host_id(get_broadcast_address());
+            slogger.info("Host {}/{} is replacing {}/{} using the same address", local_host_id, endpoint, replace_host_id, endpoint);
+        } else {
+            slogger.info("Host {}/{} is joining the token ring", local_host_id, endpoint);
```

Comment:
I don't like this "joining the token ring" is being printed on restarts, and I don't like that this entire function is called `join_token_ring`. Arguably, we only join token ring once, when we restart we're already in the ring.

But I guess we never established the precise terminology (which is bad)... So this is subjective.

Reply:
I will delete the second printout in the next version.  It's not really needed.